security/src/main/java/no/difi/meldingsutveksling/IntegrasjonspunktNokkel.java

```diff
@@ -0,0 +1,123 @@
+            throw new RuntimeException(e);
+        }
+    }
+
+    public KeyPair getKeyPair() {
+
+        KeyPair result = null;
+        try (InputStream i = openKeyInputStream()) {
+            KeyStore keystore = KeyStore.getInstance("JKS");
+            keystore.load(i, this.keystore.getPassword().toCharArray());
+            Enumeration aliases = keystore.aliases();
+            for (; aliases.hasMoreElements();) {
+                String alias = (String) aliases.nextElement();
+                boolean isKey = keystore.isKeyEntry(alias);
+                if (isKey && alias.equals(this.keystore.getAlias())) {
+                    PrivateKey key = (PrivateKey) keystore.getKey(alias, this.keystore.getPassword().toCharArray());
+                    X509Certificate c = (X509Certificate) keystore.getCertificate(alias);
+                    result = new KeyPair(c.getPublicKey(), key);
+                    break;
+                }
+            }
+            if (result == null) {
+                throw new RuntimeException("no key with alias " + this.keystore.getAlias() + " found in the keystore " +
+                        this.keystore.getPath());
+            }
+        } catch (CertificateException | KeyStoreException | IOException | NoSuchAlgorithmException | UnrecoverableKeyException e) {
+            throw new RuntimeException(e);
+        }
+        return result;
+    }
+
+    public X509Certificate getX509Certificate() {
+
+        X509Certificate result = null;
+        try (InputStream i = openKeyInputStream()) {
+            KeyStore keystore = KeyStore.getInstance("JKS");
+            keystore.load(i, this.keystore.getPassword().toCharArray());
+            Enumeration aliases = keystore.aliases();
+            while (aliases.hasMoreElements()) {
+                String alias = (String) aliases.nextElement();
+                boolean isKey = keystore.isKeyEntry(alias);
+                if (isKey && alias.equals(this.keystore.getAlias())) {
+                    result = (X509Certificate) keystore.getCertificate(alias);
+                    break;
+                }
+            }
+            if (result == null) {
+                throw new RuntimeException("no key with alias " + this.keystore.getAlias() + " found in the keystore " +
+                        this.keystore.getPath());
+            }
+        } catch (CertificateException | KeyStoreException | IOException | NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+        return result;
+    }
+
+    public SignatureHelper getSignatureHelper() {
+        try {
+            InputStream keyInputStream = openKeyInputStream();
+            return new SignatureHelper(keyInputStream, this.keystore.getPassword(), this.keystore.getAlias(), this
+                    .keystore.getPassword());
+        } catch (IOException e) {
+            throw new RuntimeException("keystore " + this.keystore.getPath() + " not found on file system.");
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00112)